### ISSUE TITLE
Should on-line edit change processed date?

### ISSUE THREAD

**Issue by Captainkirkdawson** [FreeUKGen/MyopicVicar#1580]
It has been suggested that the processing date of a batch be changed if a single record is updated.
Currently  a batch contains a number of dates.
The date it was uploaded. 
The date it was processed (as a batch).
The date it was transcribed and
The date it was last modified. This is the last time a record or the contents of the meta data about the batch changed.
All these are shown in the batch information.
In the index display of the list of batches we previously used the uploaded date; about a month ago we changed in to the processed date as it was possible for a batch to be reprocessed without being uploaded. These 3 dates are common to the batch. The date modified is the date something was changed to some part but not all of the file. The suggestion is we use this last date.

**Comment by Sherlock21** [comment]
I agree that all your present dates are needed, and should continue to be kept.
However, from the point of view of a researcher, the online edit date, or the date the file was just reprocessed, may be significant and affect a record that the Researcher took data from. So if we ( a Transcriber, of a CC or whoever, in FR terms)  has changed something within the file, then its right to show ( as Last Updated Date) that something has changed.  

OR have I missed a downside?

**Comment by SteveBiggs** [comment]
In answer to the title of the issue, I do not believe that editing a single entry should change the processed date. The processed date is when the complete batch was last processed, whereas when one entry is edited, only the 'date modified' should be updated.

It would make sense (to me) to have the date modified in the index display.

**Comment by richpomfret** [comment]
Propagate a date change to the batch, register, church and place.

**Comment by stsccfr** [comment]
We have an unbelievable 11 date fields associated with a single CSV file:

freereg1_csv_files has 6 date fields:
```
  _id: ObjectId('54638171e937902782000215'),
  file_name: 'BKMIVIMA.CSV',
  uploaded_date: ISODate('2009-08-26T12:18:00.000Z'),
  transcription_date: '24-Aug-2009',
  modification_date: '12 Dec 2017',
  updated_at: ISODate('2021-06-04T23:14:09.590Z'),
  created_at: ISODate('2014-11-12T15:49:05.663Z'),
  processed_date: ISODate('2015-03-10T18:05:10.842Z')
```

physical_files has 5 date fields
```
  _id: ObjectId('55ef5df1f493fd5e540039da'),
  file_name: 'BKMIVIMA.CSV',
  change_uploaded_date: ISODate('2009-08-26T12:18:00.000Z'),
  u_at: ISODate('2015-09-08T22:42:57.741Z'),
  c_at: ISODate('2015-09-08T22:15:13.173Z'),
  base_uploaded_date: ISODate('2009-08-26T12:18:00.000Z'),
  file_processed_date: ISODate('2015-03-10T18:05:10.842Z')
```

We are not storing any update date in the physical files, even though the CSV file does get changed when a new entry is added or an existing one is modified. 

In any case, I think the most important date to show to users is the last date that the file was changed.  This would be the date in the `modification_date` field in the `freereg1_csv_files` collection.  What we are currently displaying is actually the `file_processed_date` from the `physical_files` collection.  Why we are using the `physical_files` collection instead of `freereg1_csv_files` may be because the `processed_date` often fails to be recorded in `freereg1_csv_files` when a file is uploaded and processed.  In our current db I found over 4000 entries where processed_date is null.

So the changes here should be:

1. In the table listings of CSV files (`app/views/freereg1_csv_files/index.html.erb`) display `modification_date` from `freereg1_csv_files`.
2. Change the column header from "Processed" to "Last Modified"
3. In the screen showing the details for a selected CSV file (`app/views/freereg1_csv_files/show.html.erb`) also show this same `modification_date` field.  I would put it below the line for "Processed Date" and I would label it "Last Modified"

**Comment by sagar-ruby** [comment]
@Vino-S PR ready for review #2923.

**Comment by sagar-ruby** [comment]
Made a small update to my PR. Commit `036768a` is orhpaned and can be ignored. The current implementation 2604e95556d27dba9c956ad64a636e161c4d6f41 is in PR #2923 and ready for review.

**Comment by stsccfr** [comment]
Test on Dev and this is working nicely, so marking ready for deployment.
